docs/xxcontinuationcache.md

```diff
@@ -0,0 +1,52 @@
+<!--
+* Copyright (c) 2017, 2022 IBM Corp. and others
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution and is available at
+* https://www.eclipse.org/legal/epl-2.0/ or the Apache
+* License, Version 2.0 which accompanies this distribution and
+* is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* This Source Code may also be made available under the
+* following Secondary Licenses when the conditions for such
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] http://openjdk.java.net/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+-->
+
+# -XX:ContinuationCache
+
+![Start of content that applies to Java 21 (LTS) and later](cr/java21plus.png) This option sets the Continuation tier 1 and 2 cache size to optimize virtual thread performance.
+
+## Syntax
+
+        -XX:ContinuationCache:t1=<size>,t2=<size>
+
+| Setting                 | Value      | Default                     |
+|-------------------------|------------|:---------------------------:|
+| `t1=<size>`    | Between 0 and 2^32 (4,294,967,295) |  1                       |
+| `t2=<size>`    | Between 0 and 2^32 (4,294,967,295) |  Number of processors x 2                       |
+
+Where `t1` is tier 1 cache and `t2` is tier 2 cache.
+
+Tier 1 cache holds data that is not shareable between threads.
+
+Tier 2 cache holds data that can be shared globally between threads.
+
+## Explanation
+
+Continuation is the underlying structure or object that holds the task or state of a virtual thread. When a virtual thread is created, it is associated with a Continuation object, which holds the target task of the virtual thread. When the virtual thread starts, the VM saves the current runtime state in the internal allocated structure that is linked with the Continuation object. The VM can resume execution from that state later, typically on a different thread. When the Continuation task starts, the VM allocates native memory for the continuation to hold the internal data structure. When the Continuation task is complete, instead of freeing the allocated memory, the VM stores such a structure as a continuation cache. New Continuation objects can directly use the cached structure instead of reallocating in memory. When the VM recycles the native continuation structure that is allocated for the virtual thread, the virtual thread creation performance improves.
```

Comment:
> In this "The VM can resume execution" of what?

... resume thread execution ...

> Also, in the following which current thread are we talking about?

It is called the CarrierThread. Before the VirtualThread runs, we store the state of the current thread (CarrierThread) in the Continuation. Once the VirtualThread yields or finishes, we load the state of the previous thread from the Continuation and let it run. If the VirtualThread hasn't completed, the Continuation stores the last state of the VirtualThread before it was unmounted/stopped.

> Execution of what in the following sentence

... resume thread execution ...

> Can we move "The VM can resume execution from the saved state, typically on a different thread." after "New Continuations can reuse the cached structure instead of allocating new memory for it." i.e.

Yes, there is better flow.

> Is it ok to mention "class" in the first line ... Will it be technically correct.

No, it won't be technically correct. We create objects and classes specify the blueprint for creating objects. I would also refrain from using "object"; this is self-implied because Java is an OOP language.

Reply:
> The first line is confusing; my recommendation will be to remove it:
> 
> * `underlying structure or object`: Continuation is a class name, when instantiated it becomes an object.
> * `state of a virtual thread`: VirtualThread is also a class name. State of the VirtualThread is stored in the VirtualThread. State of the Continuation is stored in the Continuation. Both manage their states independently; a relationship can be derived between the two states.
> 
> My other recommendation will be to use `VirtualThread` instead of `virtual thread`. Then, we will be consistent with `Continuation`. Both are class names.

@babsingh - Could you please provide a qualifier for "VirtualThread"? That would make it consistent with the style used in the existing document. Else we can either use "virtual threads" and "continuation" or we can give a note to specify what "VirtualThread" and "Continuation" are. I prefer using "virtual threads" and "continuation", if it makes sense technically.

Reply:
I am also good with "virtual threads" and "continuation" because consistency is maintained.